core/kernel/tpm.c

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2020, ARM Limited.
+ */
+
+#include <kernel/generic_boot.h>
+#include <kernel/dt.h>
+#include <mm/core_mmu.h>
+#include <mm/core_memprot.h>
+#ifdef CFG_DT
+#include <kernel/dt.h>
+#include <libfdt.h>
+#endif
+#include <kernel/tpm.h>
+#include <string.h>
+
+static void *tpm_log_addr;
+static size_t tpm_log_size;
+
+/*
+ * Check whether the node at @offs contains TPM Event Log information
+ * (tpm_event_log_sm_addr and tpm_event_log_size properties) or not.
```

Comment:
Minor: not much value in repeating the property names since they stand out enough in the code IMO.